pkg/api/pod/warnings_test.go

```diff
@@ -1520,6 +1521,196 @@ func TestWarnings(t *testing.T) {
 				`spec.containers[1].ports[1]: duplicate port definition with spec.containers[1].ports[0]`,
 			},
 		},
+		{
+			name: "create duplicate container ports name in two containers",
+			template: &api.PodTemplateSpec{Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name: "foo1",
+						Ports: []api.ContainerPort{
+							{ContainerPort: 80, HostPort: 80, Protocol: api.ProtocolUDP, Name: "test"},
+						},
+					},
+					{
+						Name: "foo",
+						Ports: []api.ContainerPort{
+							{ContainerPort: 8090, HostPort: 8090, Protocol: api.ProtocolTCP, Name: "test"},
```

Comment:
> I think we should set it up so that the conflicts are always different protocol, proving that names can't be overlapped even if one is TCP and one is UDP.

In the test case for creating a pod, different protocols, TCP and UDP, are configured.

In the test case for updating a pod, the same protocol, UDP, is utilized. 

How about this? @thockin 

@thockin